File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/intro/IntroActivity.java
package sk.styk.martin.apkanalyzer.activity.intro;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.github.paolorotolo.appintro.AppIntro;
import com.github.paolorotolo.appintro.AppIntroFragment;
import com.github.paolorotolo.appintro.model.SliderPage;

import sk.styk.martin.apkanalyzer.R;
import sk.styk.martin.apkanalyzer.business.task.upload.MultipleAppDataUploadService;
import sk.styk.martin.apkanalyzer.util.FirstStartHelper;
import sk.styk.martin.apkanalyzer.util.networking.ConnectivityHelper;

/**
 * @author Martin Styk
 * @version 13.12.2017.
 */
public class IntroActivity extends AppIntro {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SliderPage analyzeAppsSlide = new SliderPage();
        analyzeAppsSlide.setTitle(getString(R.string.intro_analyze_apps));
        analyzeAppsSlide.setDescription(getString(R.string.intro_analyze_apps_description));
        analyzeAppsSlide.setBgColor(getResources().getColor(R.color.accentLight));
        analyzeAppsSlide.setImageDrawable(R.drawable.ic_lupa);

        SliderPage permissionsAppsSlide = new SliderPage();
        permissionsAppsSlide.setTitle(getString(R.string.intro_permissions));
        permissionsAppsSlide.setDescription(getString(R.string.intro_permissions_description));
        permissionsAppsSlide.setBgColor(getResources().getColor(R.color.accentLight));
        permissionsAppsSlide.setImageDrawable(R.drawable.ic_permission);

        SliderPage statisticsAppsSlide = new SliderPage();
        statisticsAppsSlide.setTitle(getString(R.string.intro_statistics));
        statisticsAppsSlide.setDescription(getString(R.string.intro_statistics_description));
        statisticsAppsSlide.setBgColor(getResources().getColor(R.color.accentLight));
        statisticsAppsSlide.setImageDrawable(R.drawable.ic_chart);

        SliderPage uploadAppsSlide = new SliderPage();
        uploadAppsSlide.setTitle(getString(R.string.intro_upload));
        uploadAppsSlide.setDescription(getString(R.string.intro_upload_description));
        uploadAppsSlide.setBgColor(getResources().getColor(R.color.accentLight));
        uploadAppsSlide.setImageDrawable(R.drawable.ic_upload);

        addSlide(AppIntroFragment.newInstance(analyzeAppsSlide));
        addSlide(AppIntroFragment.newInstance(permissionsAppsSlide));
        addSlide(AppIntroFragment.newInstance(statisticsAppsSlide));
        addSlide(AllowMetadataUploadIntroSlide.newInstance(uploadAppsSlide));

        // Hide Skip/Done button.
        showSkipButton(false);
        setProgressButtonEnabled(true);

        setVibrate(true);
        setVibrateIntensity(30);
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        // Do something when users tap on Skip button.
    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);

        // save user preferences and trigger data upload if possible
        if (currentFragment instanceof AllowMetadataUploadIntroSlide) {
            boolean isUploadAllowed = AllowMetadataUploadIntroSlide.class.cast(currentFragment).isUploadAllowed();
            ConnectivityHelper.setConnectionAllowedByUser(getApplicationContext(), isUploadAllowed);

            if (isUploadAllowed)
                MultipleAppDataUploadService.start(getApplicationContext());
        }

        FirstStartHelper.setFirstStartFinished(getApplicationContext());
        finish();
    }

    @Override
    public void onBackPressed() {
        // Do nothing
    }

    @Override
    public void onSlideChanged(@Nullable Fragment oldFragment, @Nullable Fragment newFragment) {
        super.onSlideChanged(oldFragment, newFragment);
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/ui/activity/intro/IntroActivity.kt
package sk.styk.martin.apkanalyzer.ui.activity.intro

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import com.github.paolorotolo.appintro.AppIntro
import com.github.paolorotolo.appintro.AppIntroFragment
import com.github.paolorotolo.appintro.model.SliderPage
import com.google.firebase.analytics.FirebaseAnalytics
import sk.styk.martin.apkanalyzer.util.FirstStartHelper
import sk.styk.martin.apkanalyzer.util.networking.ConnectivityHelper

/**
 * @author Martin Styk
 * @version 13.12.2017.
 */
class IntroActivity : AppIntro() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val analyzeAppsSlide = SliderPage()
        analyzeAppsSlide.title = getString(R.string.intro_analyze_apps)
        analyzeAppsSlide.description = getString(R.string.intro_analyze_apps_description)
        analyzeAppsSlide.bgColor = ContextCompat.getColor(this, R.color.accentLight)
        analyzeAppsSlide.imageDrawable = R.drawable.ic_lupa

        val permissionsAppsSlide = SliderPage()
        permissionsAppsSlide.title = getString(R.string.intro_permissions)
        permissionsAppsSlide.description = getString(R.string.intro_permissions_description)
        permissionsAppsSlide.bgColor = ContextCompat.getColor(this, R.color.accentLight)
        permissionsAppsSlide.imageDrawable = R.drawable.ic_permission

        val statisticsAppsSlide = SliderPage()
        statisticsAppsSlide.title = getString(R.string.intro_statistics)
        statisticsAppsSlide.description = getString(R.string.intro_statistics_description)
        statisticsAppsSlide.bgColor = ContextCompat.getColor(this, R.color.accentLight)
        statisticsAppsSlide.imageDrawable = R.drawable.ic_chart

        val uploadAppsSlide = SliderPage()
        uploadAppsSlide.title = getString(R.string.intro_upload)
        uploadAppsSlide.description = getString(R.string.intro_upload_description)
        uploadAppsSlide.bgColor = ContextCompat.getColor(this, R.color.accentLight)
        uploadAppsSlide.imageDrawable = R.drawable.ic_upload

        addSlide(AppIntroFragment.newInstance(analyzeAppsSlide))
        addSlide(AppIntroFragment.newInstance(permissionsAppsSlide))
        addSlide(AppIntroFragment.newInstance(statisticsAppsSlide))
        addSlide(AllowMetadataUploadIntroSlide.newInstance(uploadAppsSlide))

        // Hide Skip/Done button.
        showSkipButton(false)
        isProgressButtonEnabled = true

        setVibrate(true)
        setVibrateIntensity(30)

        FirebaseAnalytics.getInstance(this).logEvent(FirebaseAnalytics.Event.TUTORIAL_BEGIN, Bundle())
    }


    override fun onDonePressed(currentFragment: Fragment?) {
        // TODO: fix type mismatch
        super.onDonePressed(currentFragment)

        // save user preferences and trigger data upload if possible
        if (currentFragment is AllowMetadataUploadIntroSlide) {
            val isUploadAllowed = currentFragment.isUploadAllowed
            ConnectivityHelper.setConnectionAllowedByUser(applicationContext, isUploadAllowed)

//   Temporary disable uploads
//            if (isUploadAllowed)
//                MultipleAppDataUploadJob.start(applicationContext)
        }

        FirstStartHelper.setFirstStartFinished(applicationContext)
        FirebaseAnalytics.getInstance(this).logEvent(FirebaseAnalytics.Event.TUTORIAL_COMPLETE, Bundle())
        finish()
    }

    override fun onBackPressed() {
        // Do nothing
    }
}
